Sources/Testing/Running/XCTestScaffold.swift

```diff
@@ -163,7 +163,8 @@ public enum XCTestScaffold: Sendable {
       // if the issue represents an assertion failure or a thrown error.
       if !issue.isKnown {
         let sourceLocation = issue.sourceLocation ?? .init()
-        XCTFail(String(describing: issue), file: sourceLocation._filePath, line: UInt(sourceLocation.line))
+
+        recordFailure(inTest: testCase, withDescription: String(describing: issue), inFile: sourceLocation._filePath, atLine: sourceLocation.line, expected: false)
```

Comment:
I think you'd want to pass `true` for `expected:`.

And this predates this PR, but if `issue.sourceLocation` is `nil`, I think it should not default to `?? .init()` because that would capture the source location inside `XCTestScaffold.swift`. Instead, I think it should use `inFile: "<unknown>, atLine: 0`. Those are the traditional values representing "no source location known".